File: 2372-rearrange-characters-to-make-target-string/rearrange-characters-to-make-target-string.py
class Solution:
    def rearrangeCharacters(self, s: str, target: str) -> int:
        if len(target)>len(s):
            return 0

        freq = Counter(s)
        freq_target = Counter(target)
        
        count = math.inf
        
            
        for letter in target:
            a=freq_target[letter]
            if letter in freq:
                b=freq[letter]
            else:
                return 0
            count = min(count,b//a)
        return count

File: 2372-rearrange-characters-to-make-target-string/rearrange-characters-to-make-target-string.js
/**
 * @param {string} s
 * @param {string} target
 * @return {number}
 */
var rearrangeCharacters = function(s, target) {
    let freq_s = {}
    let freq_target = {}

    for (letter of target){
        freq_target[letter] = (freq_target[letter] || 0) + 1
        
    }
    for (letter of s){
        if (letter in freq_target){
            freq_s[letter] = (freq_s[letter] || 0)+1
        }
    }
    if (Object.keys(freq_s).length!==Object.keys(freq_target).length){
        return 0
    }
    let minimum = Infinity
    for (key of Object.keys(freq_s)){
        if (freq_s[key]<freq_target[key]){
            return 0
        }
        minimum = Math.min(minimum,Math.floor(freq_s[key]/freq_target[key]))
    }
    return minimum
};